### PR TITLE
add port number support

### PR DESCRIPTION
**Pull request by msivasubramaniaan**
Fixes: https://github.com/redhat-developer/vscode-openshift-tools/issues/2781

### CODE REVIEW

File: src/webview/common/componentNameInput.tsx [hunk 1]
```diff
@@ -10,24 +10,46 @@ export type ComponentNameInputProps = {
...
+                value={props.portNumber}
+                error={!props.isPortNumberFieldValid}
+                helperText={props.portNumberErrorMessage}
+                onChange={(e) => {
+                    window.vscodeApi.postMessage({
+                        action: 'validatePortNumber',
+                        data: e.target.value
+                    });
+                    props.setPortNumber(parseInt(e.target.value, 10));
+                }} />
```
Comment (datho7561) [inline src/webview/common/componentNameInput.tsx:52]: This component is for the component name input field. Please do one of the following:
- Move the code for the port input field into its own component
- Rename the component to reflect that it also contains the port input field

File: src/webview/common-ext/createComponentHelpers.ts [hunk 2]
```diff
@@ -109,6 +109,33 @@ export function validateComponentName(name: string): string {
...
+/**
+ * Returns the validation message if the component name is invalid, and undefined otherwise.
+ *
+ * @param name the port number to validate
+ * @returns the validation message if the component name is invalid, and undefined otherwise
+ */
+export function validatePortNumber(portNumber: string): string {
+    let validationMessage: string;
+    if (portNumber.trim().length > 0) {
+        validationMessage = NameValidator.lengthName(
```
Comment (datho7561) [inline src/webview/common-ext/createComponentHelpers.ts:121]: If I want to set the port to `20`, then I have to enter `00020`. Is it possible to validate the port based only on the value of the number?

**Review by datho7561 — CHANGES_REQUESTED** [review]

**Comment by msivasubramaniaan** [comment]
@datho7561 Thanks for your review. I just addressed the review comments and pushed the code. Can you please have a look on the latest code?

**Review by datho7561 — APPROVED** [review]
Looks good, and works well. Thanks, Muthu!

**Comment by datho7561** [comment]
It seems like there are linter errors

**Comment by msivasubramaniaan** [comment]
> It seems like there are linter errors

Fixed lint errors

**Comment by mohitsuman** [comment]
@msivasubramaniaan can you please share the gif/video of how it behaves.

**Comment by datho7561** [comment]
Here's what it looks like:

Screenshot from 2023-09-21 08-53-28

**Comment by datho7561** [comment]
The integration tests seem to be failing.

**Comment by msivasubramaniaan** [comment]
@mohitsuman Here is the demo gif:
port-number

**Comment by datho7561** [comment]
Currently, you can't set the port for components created from template or imported from git

**Review by datho7561 — APPROVED** [review]
Looks good! Feel free to merge after rebasing

**Comment by msivasubramaniaan** [comment]
@datho7561 Thanks for your review.

**Pull request merged**